src/irr/asset/IAssetManager.cpp

```diff
@@ -2,36 +2,155 @@
 #include "irr/asset/CGeometryCreator.h"
 #include "irr/asset/CMeshManipulator.h"
 
+
+#ifdef _IRR_COMPILE_WITH_X_LOADER_
+#include "irr/asset/CXMeshFileLoader.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_OBJ_LOADER_
+#include "irr/asset/COBJMeshFileLoader.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_STL_LOADER_
+#include "irr/asset/CSTLMeshFileLoader.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_PLY_LOADER_
+#include "irr/asset/CPLYMeshFileLoader.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_BAW_LOADER_
+#include "irr/asset/CBAWMeshFileLoader.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_BMP_LOADER_
+#include "irr/asset/CImageLoaderBMP.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_DDS_LOADER_
+#include "irr/asset/CImageLoaderDDS.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_JPG_LOADER_
+#include "irr/asset/CImageLoaderJPG.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_PNG_LOADER_
+#include "irr/asset/CImageLoaderPNG.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_TGA_LOADER_
+#include "irr/asset/CImageLoaderTGA.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_STL_WRITER_
+#include "irr/asset/CSTLMeshWriter.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_PLY_WRITER_
+#include "irr/asset/CPLYMeshWriter.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_BAW_WRITER_
+#include"irr/asset/CBAWMeshWriter.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_TGA_WRITER_
+#include "irr/asset/CImageWriterTGA.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_JPG_WRITER_
+#include "irr/asset/CImageWriterJPG.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_PNG_WRITER_
+#include "irr/asset/CImageWriterPNG.h"
+#endif
+
+#ifdef _IRR_COMPILE_WITH_BMP_WRITER_
+#include "irr/asset/CImageWriterBMP.h"
+#endif
+
 using namespace irr;
 using namespace asset;
 
-std::function<void(IAsset*)> irr::asset::makeAssetGreetFunc(const IAssetManager* const _mgr)
+std::function<void(SAssetBundle&)> irr::asset::makeAssetGreetFunc(const IAssetManager* const _mgr)
```

Comment:
@Crisspl check if the asset writers on your branch with non-ptr caches still work!